src/PipelineNet/MiddlewareResolver/MiddlewareResolverResult.cs

```diff
@@ -12,10 +12,9 @@ public class MiddlewareResolverResult
 
         /// <summary>
         /// Gets or sets the value indicating whether the middleware should be disposed.
-        /// Set this to <see langword="true"/> if the middleware is IDisposable or
-        /// IAsyncDisposable (requires .NET Standard 2.1 or greater)
-        /// and is not disposed by a dependency injection container.
+        /// Set this to <see langword="true"/> if the middleware is not disposed
+        /// by a dependency injection container.
         /// </summary>
-        public bool IsDisposable { get; set; }
+        public bool Dispose { get; set; }
```

Comment:
Can you please introduce a new property and leave the `IsDisposable` one with an `Obsolete` attribute instead?

I'll push a new version for 0.x with all these changes in the master branch, then for version 1.0 we can drop the `IsDisposable` completely, same will happen for the `Finally` method.

Reply:
I would prefer not to. This should only affect people with recent custom `IMiddlewareResolver` implementations and there is no way to know whether to `Dispose()` middleware when properties are set to different values.

Reply:
We can keep the `IsDisposable` as a getter/setter marked as obsolete for now. Something like:


```csharp

         public bool Dispose { get;set;}

        [Obsolete("This attribute will be removed in next major version. Use Dispose instead.")]
        public bool IsDisposable
        {
            get { return Dispose; }
            set { Dispose = value; }
        }

```

Reply:
Ok, I added it.

Reply:
BTW this is how finally branching can be removed without obsolete APIs:
```C#
public class AsyncResponsibilityChain<TParameter, TReturn>
{
    private Type _finallyType = new DefaultAsyncFinally();

    public async Task<TReturn> Execute(TParameter parameter, CancellationToken cancellationToken)
    {
        if (index == MiddlewareTypes.Count)
        {
            finallyResolverResult = MiddlewareResolver.Resolve(_finallyType);
            EnsureMiddlewareNotNull(finallyResolverResult, _finallyType);
            next = async (p) => await RunFinallyAsync(finallyResolverResult, p, cancellationToken).ConfigureAwait(false);
        }
    }

    private class DefaultAsyncFinally : ICancellableAsyncFinally<TParameter, TReturn>
    {
        public async Task<TReturn> Finally(TParameter parameter, CancellationToken cancellationToken) =>
            await Task.FromResult(default(TReturn)).ConfigureAwait(false);
    }
}
```